From 88e5da67c18cbdb03c88994483daf7a41005c9b0 Mon Sep 17 00:00:00 2001
From: Yael Samet <yael.samet@intel.com>
Date: Mon, 7 Jan 2019 09:01:24 +0200
Subject: [PATCH 1272/1417] mei: dal: dal_dev_read() - check for invalid
 argument

Need to dal_dev_read() if user the supplied buffer is not NULL.

Change-Id: I83ce529d9859090ddc959812a3f9c3abd5c376dd
Signed-off-by: Yael Samet <yael.samet@intel.com>
Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
---
 drivers/misc/mei/dal/dal_cdev.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/misc/mei/dal/dal_cdev.c b/drivers/misc/mei/dal/dal_cdev.c
index 48f209a..7b1aa55 100644
--- a/drivers/misc/mei/dal/dal_cdev.c
+++ b/drivers/misc/mei/dal/dal_cdev.c
@@ -107,6 +107,9 @@ static ssize_t dal_dev_read(struct file *fp, char __user *buf,
 	size_t r_len, len;
 	unsigned int copied;
 
+	if (!buf)
+		return -EINVAL;
+
 	ret = dal_wait_for_read(dc);
 	if (ret)
 		return ret;
-- 
2.7.4

